powerspikegg/go_gateway/gateway_test.go

```diff
@@ -0,0 +1,172 @@
+		s.Serve(l)
+	}(mock.server, lis)
+	return mock, nil
+}
+
+// Mock of the match endpoint
+func (s *mockMatchFetcherServer) Match(_ context.Context, req *fetcherpb.MatchRequest) (*lolpb.MatchReference, error) {
+	s.matchRequests = append(s.matchRequests, req)
+
+	if s.err != nil {
+		return nil, s.err
+	}
+
+	return s.response, nil
+}
+
+// Resets the mock requests sent list
+func (s *mockMatchFetcherServer) reset() {
+	s.matchRequests = nil
+	s.summonerRequests = nil
+}
+
+// Mock of the match endpoint
+func (s *mockMatchFetcherServer) UpdateSummoner(req *lolpb.Summoner, resp fetcherpb.MatchFetcher_UpdateSummonerServer) error {
+	s.summonerRequests = append(s.summonerRequests, req)
+
+	if s.err != nil {
+		return s.err
+	}
+
+	resp.Send(s.response)
+
+	return nil
+}
+
+// TestSummonerCommand ensures summoner command correctly recovers summoners' matchs from the fetcher
+func TestGateway(t *testing.T) {
+	tt := []struct {
+		name             string
+		serverRequest    string
+		serverResponse   *lolpb.MatchReference
+		expectedResponse string
+		expectedStatus   int
+	}{
+		{
+			name:          "summoner query",
+			serverRequest: "/api/summoner/Rangork/EUW",
+			serverResponse: &lolpb.MatchReference{
+				// FIXME: weird to only have a single match possibility here, we should make two test structs,
+				// one for each route as summoner needs multiple match response in its stream eventually
+				Id: 3122561986,
+			},
+			expectedResponse: "{result: [{\"id\":3.122561986e+09}]}", // FIXME
+			expectedStatus:   http.StatusOK,
+		},
+		{
+			name:          "match query",
+			serverRequest: "/api/match/3122561986/EUW",
+			serverResponse: &lolpb.MatchReference{
+				Id: 3122561986,
+			},
+			expectedResponse: "{\"id\":3.122561986e+09}", // FIXME
+			expectedStatus:   http.StatusOK,
+		},
+	}
+
+	matchFetcherServer, err := newMockMatchFetcherServer()
+
+	defer matchFetcherServer.server.Stop()
+	if err != nil {
+		t.Fatalf("unable to create mock match fetcher server: %v", err)
+	}
+
+	//computationClient := newMockMatchComputationServer() // TODO(archangel): computation tests
+
+	conn, err := grpc.Dial(matchFetcherServer.address, grpc.WithInsecure())
+	if err != nil {
+		t.Fatalf("unable to connect to the mock match fetcher server: %v", err)
+	}
+	defer conn.Close()
+
+	matchFetcherClient := fetcherpb.NewMatchFetcherClient(conn)
+	//computationClient := computation.NewMatchComputationClient(conn) // TODO(archangel): computation tests
+
+	lis, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		t.Fatalf("failed to listen: %v", err)
+	}
+
+	s := createServer(matchFetcherClient, nil)
+
+	go func(s *http.Server, l net.Listener) {
+		s.Serve(l)
+		fmt.Printf("creating server on: %s", s.Addr)
+	}(s.server, lis)
+
+	for _, testValue := range tt {
+		matchFetcherServer.reset()
+		matchFetcherServer.response = testValue.serverResponse
+
+		// Create a request to pass to our handler. We don't have any query parameters for now, so we'll
```

Comment:
you can remove this comment. This is pretty obvious, and if someone does not get the `nil`, he'll have to go through the doc.